File: code/java/majorityElement.java
class Solution {
    public int majorityElement(int[] nums) {
        if(nums.length == 1)
            return nums[0];
        
        Arrays.sort(nums);
        System.out.println(Arrays.toString(nums));
        int res = nums[0];
        int count = 1;
        int temp = 0;
        for(int i=1;i<nums.length;i++)
            if(nums[i]==nums[i-1])
                temp++;
            else{
                if(count<temp){
                    count = temp;
                    res = nums[i-1];
                }
                temp = 1;
            }
        if(count < temp)
            return nums[nums.length-1];
        
        return res;

    }
}

File: code/python/majorityElement.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        nums.sort()
        if len(nums) == 1:
            return nums[0]
        res = nums[0]
        count = 1
        temp = 1
        
        for i in range(1,len(nums)):
            if nums[i] == nums[i-1]:
                temp += 1
            else:
                if count < temp:
                    count = temp
                    res = nums[i-1]
                temp = 1
        
        if count < temp:
            res = nums[-1]
        return res
